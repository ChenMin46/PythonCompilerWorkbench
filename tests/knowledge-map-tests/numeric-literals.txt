3
3.0
3.14
-3
-3.14
3.14159
3.
-3.
