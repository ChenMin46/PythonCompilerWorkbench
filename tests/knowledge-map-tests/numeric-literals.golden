3
3.0
3.14
-3
-3.14
3.14159
3.
-3.
===
Module
  body: [
    Expr
      value: Num >> '3'
        n: 3
    Expr
      value: Num >> '3.0'
        n: 3.0
    Expr
      value: Num >> '3.14'
        n: 3.14
    Expr
      value: Num >> '-3'
        n: -3
    Expr
      value: Num >> '-3.14'
        n: -3.14
    Expr
      value: Num >> '3.14159'
        n: 3.14159
    Expr
      value: Num >> '3.'
        n: 3.0
    Expr
      value: Num >> '-3.'
        n: -3.0
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "3"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "3.0"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "3.14"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "-3"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "-3.14"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "3.14159"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "3."
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "-3."
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
